File: app/src/main/java/com/lxk/animandcustomviewdemo/animator/evaluator/ReverseEvaluator.java
package com.lxk.animandcustomviewdemo.animator.evaluator;

import android.animation.TypeEvaluator;

/**
 * @author https://github.com/103style
 * @date 2019/12/11 13:49
 * 将属性动画的值 整形化
 */
public class ReverseEvaluator implements TypeEvaluator<Integer> {

    @Override
    public Integer evaluate(float fraction, Integer startValue, Integer endValue) {
        return (int) (endValue - fraction * (endValue - startValue));
    }
}

File: app/src/main/java/com/lxk/animandcustomviewdemo/animator/evaluator/ReverseEvaluator.kt
package com.lxk.animandcustomviewdemo.animator.evaluator

import android.animation.TypeEvaluator

/**
 * @author https://github.com/103style
 * @date 2020/1/10 14:50
 * 将属性动画的0 - 1值 整形化为起始和结束之间的值
 */
class ReverseEvaluator : TypeEvaluator<Int> {
    override fun evaluate(fraction: Float, startValue: Int, endValue: Int): Int {
        return (endValue - fraction * (endValue - startValue)).toInt()
    }
}